### PR TITLE
GH-33830: Clarify handling of Null values in REE encoding

### LINKED ISSUE #33830: Clarify handling of Null values in REE encoding
In the Columnar format, the handling of null values in the REE type is not described properly.

- The null count of the parent array should be 0, since the parent has no validity bitmap (and there is a general assumption in arrow code that a non-zero null count means the standard validity bitmap is present)
- The document only indirectly describes how Null values are encoded

Thanks @tustvold for catching this!

[1] [https://github.com/apache/arrow/pull/13333#discussion_r1083470362](https://github.com/apache/arrow/pull/13333#discussion_r1083470362)

### Component(s)

Documentation, Format

### PR DESCRIPTION
**Pull request by zagto**
* Closes: #33830

### CODE REVIEW

File: docs/source/format/Columnar.rst [hunk 1]
```diff
@@ -815,6 +815,9 @@ A run must have have a length of at least 1. This means the values in the
 run ends array all are positive and in strictly ascending order. A run end cannot be
 null.
 
+The REE parent has no validity bitmap, and it's null count field should always be 0.
+validity bitmap. Null values are encoded as runs with the value null.
```
Comment (westonpace) [inline docs/source/format/Columnar.rst:819]:
```suggestion
Null values are encoded as runs with the value null.
```

Comment (zagto) [reply docs/source/format/Columnar.rst]: Thanks! fixed

**Review by tustvold — APPROVED** [review]

**Review by westonpace — APPROVED** [review]
This matches my understanding (parent array has no buffers and thus, no null count)

**Review by zeroshade — APPROVED** [review]

**Review by alamb — COMMENTED** [review]
Thank you @zagto

**Pull request merged**